tests/pytests/test_cert_framework_basic.py

```diff
@@ -76,6 +76,45 @@ def test_cf_cc_trust_data():
 
     libcf.delete_cc_trust_data(cctd)
 
+# ##############################################################################
+def test_cf_cc_trust_data_add_or_update_new_domain():
+    """
+    Exercise add_or_update_new_domain(), which will create a new certified_domain()
+    object. Dismantiling this cc_trust_data() will test the destructor of that
+    object which should correctly release memory for new certified-domains.
+    """
+    cctd = libcf.new_cc_trust_data()
+
+    result = libcf.cc_trust_data_add_or_update_new_domain(cctd,
+                                                          'test-security-domain',
+                                                          'test-dummy-certificate',
+                                                          'localhost', 8121,
+                                                          'localhost', 8123)
+    assert result is True
+
+    libcf.delete_cc_trust_data(cctd)
```

Comment:
This delete call in new test case will invoke the `cc_trust_data()` destructor.

I didn't quite find a way to confirm or check thru pytests that there is a memory leak when the new C++ code in the destructor was not added.

But from code inspection, the memory release code in the desctructor is fine, so I'm ok with just this exerciser unit test.